TESTS/mbed_hal/watchdog_reset/watchdog_reset_tests.h

```diff
@@ -0,0 +1,79 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @addtogroup hal_watchdog_tests
+ * @{
+ */
+
+#ifndef MBED_WATCHDOG_RESET_TESTS_H
+#define MBED_WATCHDOG_RESET_TESTS_H
+
+#if DEVICE_WATCHDOG
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**Test the Watchdog HAL API
+ *
+ * @note The reason all test cases below are handled by one test function
+ * is that the host side provides all the test logic. DUT simply handles
+ * host's commands.
+ *
+ * Test 1.
+ * Given a device supporting a Watchdog API.
+ * When the watchdog is started.
+ * Then it does not reset the device until timeout expires.
+ * When the timeout expires.
+ * Then the device is restarted.
+ *
+ * Test 2.
+ * Given a device with watchdog started.
+ * When the watchdog is kicked before timeout expires.
+ * Then the device restart is prevented.
+ * When the watchdog is *NOT* kicked again before next timeout expires.
+ * Then the device is restarted.
+ *
+ * Test 3.
+ * Given a device supporting sleep mode, with watchdog started.
+ * When the device is in sleep mode and watchdog timeout expires.
+ * Then the device is restarted.
+ *
+ * Test 4.
+ * Given a device supporting deepsleep mode, with watchdog started.
+ * When the device is in deepsleep mode and watchdog timeout expires.
+ * Then the device is restarted.
+ *
+ * Test 5.
+ * Given a device supporting 'disable_watchdog' feature, with watchdog started.
+ * When the watchdog is stopped before timeout expires.
+ * Then the device is not restarted.
+ * When the watchdog is started again and it's timeout expires.
+ * Then the device is restarted.
+ */
+void test_watchdog_operation();
```

Comment:
Since all the logic is provided by the host side all these test cases are handled same way on the dev side -- by simply handling the commands. I feel they should get separated anyway to be consistent with other test suits. What would be the best way to do that? Copy the code into separate test suits by e.g. `TESTS/mbed_hal/{watchdog_reset,watchdog_kick,watchdog_sleep,watchdog_deepsleep,watchdog_disable}`? This would introduce code duplication on dev and host side.